src/app/src/components/VectorTileFacilityGridLayer.jsx

```diff
@@ -0,0 +1,164 @@
+        performingNewSearch,
+        storedResetCount,
+        setStoredResetCount,
+        resetButtonClickCount,
+    ]);
+
+    return vectorTileURLWithQueryParams;
+}
+
+const VectorTileFacilityGridLayer = ({
+    tileURL,
+    fetching,
+    resetButtonClickCount,
+    tileCacheKey,
+    getNewCacheKey,
+    handleCellClick,
+    minZoom,
+    maxZoom,
+}) => {
+    const vectorTileURL = useUpdateTileURL(
+        tileURL,
+        fetching,
+        resetButtonClickCount,
+        getNewCacheKey,
+    );
+
+    const vectorTileLayerRef = useRef(null);
+
+    if (!tileCacheKey) {
+        // We throw an error here if the tile cache key is missing.
+        // This crashes the map, intentionally, but an ErrorBoundary
+        // handles the crash so that the application continues to run.
+        throw new Error('Missing tile cache key');
+    }
+
+    return (
+        <VectorGrid
+            ref={vectorTileLayerRef}
+            key={vectorTileURL}
+            url={vectorTileURL}
+            type="protobuf"
+            rendererFactory={L.canvas.tile}
+            updateWhenZooming={false}
+            minZoom={minZoom}
+            maxZoom={maxZoom}
+            vectorTileLayerStyles={{
+                facilitygrid(properties) {
+                    const layer = get(
+                        vectorTileLayerRef,
+                        'current.leafletElement',
+                    );
+                    // eslint-disable-next-line no-underscore-dangle
+                    const zoomLevel = layer._map.getZoom();
+                    const factor = 2 * Math.max(0, zoomLevel - 4);
+                    const count = get(properties, 'count', 0);
+                    if (count === 0) {
+                        return {
+                            fill: false,
+                            stroke: false,
+                        };
+                    }
+                    const fillColor = GRID_COLOR_RAMP.reduce(
+                        // eslint-disable-next-line no-confusing-arrow
+                        (color, [val, c]) => count + factor > val ? c : color, GRID_COLOR_RAMP[0],
+                    );
+
+                    return {
+                        fill: true,
+                        fillColor,
+                        fillOpacity: 0.8,
+                        stroke: true,
+                        weight: 0.5,
+                        color: '#0427A4',
+                    };
+                },
+            }}
+            subdomains=""
+            zIndex={100}
+            interactive
+            onClick={handleCellClick}
+        />
+    );
+};
+
+VectorTileFacilityGridLayer.propTypes = {
+    handleCellClick: func.isRequired,
+    tileURL: string.isRequired,
+    tileCacheKey: string.isRequired,
+    fetching: bool.isRequired,
+    resetButtonClickCount: number.isRequired,
+};
+
+const createURLWithQueryString = (qs, key) =>
+    `/tile/facilitygrid/${key}/{z}/{x}/{y}.pbf`.concat(isEmpty(qs) ? '' : `?${qs}`);
```

Comment:
Was thinking that it might be possible to extract this function out into a utility since it'll soon occur in two places. However, I also don't think the function should change much in the future -- even if the qs is encoded in the cache key by the server -- so it's probably better and cheaper just to leave as is.

Reply:
👍